src/SeoToolkit.Umbraco.Redirects.Core/Controllers/RedirectsController.cs

```diff
@@ -134,5 +144,67 @@ public IActionResult Delete(DeleteRedirectsPostModel postModel)
             _redirectsService.Delete(postModel.Ids);
             return GetAll(1, 20);
         }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+        public IActionResult Validate(ImportRedirectsFileExtension fileExtension, string domain, IFormFile file)
+        {
+            if (file.Length == 0)
+            {
+                return BadRequest("Please select a file");
+            }
+        
+            using var memoryStream = new MemoryStream();
+            file.CopyTo(memoryStream);
+
+            var result = _redirectsImportHelper.Validate(fileExtension, memoryStream, domain);
+            if (result.Success)
+            {
+                if (_httpContextAccessor.HttpContext is null)
```

Comment:
Do you need the _httpContextAccessor here? You are able to access HttpContext directly because of the controller right?

Reply:
Will check, thanks!

Reply:
Removed !